### PR TITLE
support non-directory/non-zip checkpoint names

### PR DESCRIPTION
**Pull request by dpressel**
sometimes there might be multiple pid names in
the same file. this happens when multiple runs are
done and then the user needs to load only one.

by checking the job status or even just the time
of each checkpoint written out, the user can easily
determine the PID-specific job name, but its not
currently loadable.

This change allows the user to give a longer expression
as the checkpoint name and resolves that to the full
checkpoint name, and passes this along to the suffix

### CODE REVIEW

File: baseline/utils.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def save_vocabs(basedir, embeds_or_vocabs, name='vocabs'):
 
 
 @export
-def load_vocabs(directory):
+def load_vocabs(directory: str, suffix: Optional[str] = None):
     vocab_fnames = find_files_with_prefix(directory, 'vocabs')
```
Comment (sagnik) [inline baseline/utils.py:328]: wouldnt you pass in `suffix` here?

File: baseline/pytorch/torchy.py [hunk 2]
```diff
@@ -85,29 +85,3 @@ def append2seq(seq, modules):
 
     for i, module in enumerate(modules):
         seq.add_module('%s-%d' % (str(module).replace('.', 'dot'), i), module)
-
```
Comment (sagnik) [inline baseline/pytorch/torchy.py:88]: are these changes intended?

Comment (dpressel) [reply baseline/pytorch/torchy.py:88]: yeah, just culling some unused code

File: baseline/utils.py (hunk 1 above)
Comment (dpressel) [reply baseline/utils.py]: oops you are right. updated
